pkg/workloadmeta/types.go

```diff
@@ -554,6 +554,26 @@ type Event struct {
 	Entity  Entity
 }
 
+// SubscriberPriority is a priority for subscribers to the store.  Subscribers
+// are notified in order by their priority, with each notification blocking the
+// next, so this allows control of which compoents are informed of changes in
+// the store first.
+type SubscriberPriority int
+
+const (
+	// TaggerPriority is the priority for the Tagger.  The Tagger must always
+	// come first.
+	TaggerPriority SubscriberPriority = iota
+
+	// ADPriority is the priority for autodiscovery.  This must come after
+	// Tagger, but before other components.
```

Comment:
> but before other components.

I don't think so, mind elaborating more on this? 

I don't think we need a separate priority for AD and imho we should add it only when it's needed.

Reply:
I did this because other components (logs-agent in particular) depend on both Tagger and AD.  If you're confident that the ordering is not critical, I'm happy to remove this.

Reply:
the logs agent discovers containers only via AD, right? so it's enough to have the tagger more prioritised than AD, to make sure the tagger knows the container when the logs agent queries the tags of a container

Reply:
Sounds good, thanks!